packages/vitest/src/node/mocker.ts

```diff
@@ -171,12 +170,12 @@ export class VitestMocker {
 
     if (Array.isArray(obj))
       return []
-    else if (type !== 'Object' && type !== 'Module')
+    else if ((type !== 'Object' && type !== 'Module') || !this.spy)
```

Comment:
Should `!this.spy` here be a logic error instead of a silent check?

Reply:
agree, this should never happen